### PR TITLE
Allow user to set WindowConf and hints

### PR DESCRIPTION
**Pull request by bch29**
The library becomes much more useful practical applications if the user can set the width/height/title of the window. Also, I wanted to make a window that can't be resized, so I added the ability for the user to add hints as well.

### CODE REVIEW

**Comment by tobbebex** [comment]
What happens if the extra hints conflicts with those generated from the context format?

**Comment by bch29** [comment]
Hmm, you're right. I could document it, check that none of the hints the user provides can collide, and throw an exception if they do.

**Comment by bch29** [comment]
I've made that change now. The other option would be to make our own datatype for 'extra' hints and map it to the GLFW ones. What do you think?

**Comment by tobbebex** [comment]
I'll leave that decision to @plredmond, either way works for me! With this solution, you wouldn't at least trick GPipe into thinking the context has a format that it doesn't which would have given strange behaviours. Now you "just" get a runtime error before the GPipe code even runs. Your proposed solution would get rid of that too though.

File: GPipe-GLFW/stack.yaml [hunk 1]
```diff
@@ -1,8 +1,5 @@
 flags: {}
 packages:
 - '.'
-extra-deps:
-- GPipe-2.0
-- GLFW-b-1.4.7.3
-- bindings-GLFW-3.1.1.4
-resolver: lts-3.1
+extra-deps: []
+resolver: nightly-2015-10-31
```
Comment (plredmond) [inline GPipe-GLFW/stack.yaml]: Can we use an LTS instead of nightly?

**Comment by plredmond** [comment]
I don't think it's necessary to add a datatype to map between allowed hints yet. You're welcome to add it to this PR if you want, though.

File: GPipe-GLFW/stack.yaml (hunk 1 above)
Comment (tobbebex) [inline GPipe-GLFW/stack.yaml]: GPipe isn't in an LTS until 4.0.

Comment (plredmond) [inline GPipe-GLFW/stack.yaml]: My mistake.

Comment (tobbebex) [inline GPipe-GLFW/stack.yaml]: But I wonder how much longer that will be...

Comment (bch29) [inline GPipe-GLFW/stack.yaml]: I could always change it back to LTS and add GPipe as an extra-dep. I don't think the stack.yaml used here matters too much though -- users are going to get the library through stackage/hackage anyway, not use this stack.yaml at all.

Comment (plredmond) [inline GPipe-GLFW/stack.yaml]: You're right. I'm testing now and will probably merge today

Comment (bch29) [inline GPipe-GLFW/stack.yaml]: Cool, thanks. Let me know if there are any problems -- I only did a small amount of testing on my side, but it's working for my project perfectly fine.

**Pull request merged**